server/src/attestation.rs

```diff
@@ -0,0 +1,257 @@
+use crate::users::policy::*;
+use anyhow::Result;
+use attestation_service::Service as AS;
+use std::path::PathBuf;
+use std::sync::Arc;
+use tokio::sync::RwLock;
+use tonic::{Request, Response, Status};
+extern crate serde;
+use self::serde::{Deserialize, Serialize};
+use tonic::transport::Server;
+
+use crate::attestation_api::attestation_service_server::{
+    AttestationService, AttestationServiceServer,
+};
+use crate::attestation_api::{AttestationRequest, AttestationResponse};
+
+const DEFAULT_ATTESTATION_SOCK: &str = "127.0.0.1:3000";
+
+// The evidence(.json) must align with the "attestation" definition in:
+// https://github.com/confidential-containers/kbs/blob/main/docs/kbs_attestation_protocol.md#attestation
+// The evidence is deserializes by Attestation Server partially, and it will be deserialized wholly by Attesattion Service Crate.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Evidence {
+    // The evidence's TEE type.
+    pub tee: String,
+}
+
+#[derive(Debug)]
+pub struct Service {
+    locker: Arc<RwLock<u8>>,
```

Comment:
Rust locking APIs are designed to associate a  lock/mutex with the data it protects. What you're doing here is closer to a C-ish approach, and I think this calls for a slightly different pattern that would look like that:

```Rust
struct InnerService {
     workdir: PathBuf,
}

pub struct Service {
     service: Arc<RwLock<InnerService>>,
}

impl Service {
    pub fn new(workdir: PathBuf) -> Self {
        Self {
            service: Arc::new(RwLock::new(InnerService { workdir })),
        }
    }
}
```

With that pattern the lock becomes transparent for anyone using `Service`, and you have the guarantee that one must take the lock if it wants to use the actual `Service` data, even when/if we expand the `Service` API or structure in the future.

Also callers of the API don't need to create an artificial lock, that's handled internally.

Reply:
OK, I will update to make the policy/reference_data files' operation interfaces as separate service which will be wrapped inside a RW lock.